src/app/components/media/Media.js

```diff
@@ -37,6 +37,78 @@ const MediaContainer = Relay.createContainer(MediaComponent, {
           quote
           embed_path
         }
+        comments: annotations(first: 10000, annotation_type: "comment") {
+          edges {
+            node {
+              ... on Comment {
+                id
+                dbid
+                text
+                parsed_fragment
+                annotator {
+                  id
+                  name
+                  profile_image
+                }
+                comments: annotations(first: 10000, annotation_type: "comment") {
+                  edges {
+                    node {
+                      ... on Comment {
+                        id
+                        created_at
+                        text
+                        annotator {
+                          id
+                          name
+                          profile_image
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+        clips: annotations(first: 10000, annotation_type: "clip") {
+          edges {
+            node {
+              ... on Dynamic {
+                id
+                data
+                parsed_fragment
+              }
+            }
+          }
+        }
+        tags(first: 10000) {
+          edges {
+            node {
+              id
+              dbid
+              fragment
+              parsed_fragment
+              annotated_id
+              annotated_type
+              annotated_type
+              tag_text_object {
+                id
+                text
+              }
+            }
+          }
+        }
+        geolocations: annotations(first: 10000, annotation_type: "geolocation") {
+          edges {
+          node {
+              ... on Dynamic {
+                id
+                parsed_fragment
+                content
+              }
+            }
+          }
+        }
```

Comment:
A good performance improvement we could do here in the future would allow us to make only one query for annotations instead of four:
```
annotations(first: 10000, annotation_type: ['geolocation', 'clip', 'comment', 'tag']) {
  edges {
    node {
      ... on Dynamic { ... }
      ... on Comment { ... }
      ... on Tag { ... }
    }
  }
}
```
But then the API should also accept a list of `annotation_type` as argument: https://github.com/meedan/check-api/blob/develop/app/graph/mutations/graphql_crud_operations.rb#L551

Reply:
just be careful with the configs/optimistic updates on mutations

Reply:
this is for later because requires changes on the backend, but would be great to move them from the main query to the timeline components as fragments, if possible